ethereum/contracts/ETHApp.sol

```diff
@@ -107,7 +109,9 @@ contract ETHApp is RewardSource, AccessControl {
         external
         override
         onlyRole(REWARD_ROLE)
+        nonReentrant()
```

Comment:
why is nonReentrant used/necessary here?

Reply:
The usage of the call function forwards all remaining gas. So this might be vulnerable to re-entrancy attack.
Please refer link for more info : https://medium.com/coinmonks/protect-your-solidity-smart-contracts-from-reentrancy-attacks-9972c3af7c21

Reply:
yeh, adding this makes sense as a general default practice, however I don't think we should blindly follow practices without investigating and confirming 100% whether or not they are applicable to our code. Similarily, whenever there is some uncertainty, ie _"this might be vulnerable"_, we should flag that and resolve the uncertainty to be sure exactly whether or not there is an issue, rather than just making an assumption and using a practice

In this case, I don't believe there is a re-entrancy attack vector, specifically because:
 - the function is permissioned - only a specific contract which we already know the code for can call this, and that contract cannot re-enter
 - even if for some reason that contract could re-enter, or another contract was used that could, there is still no attack vector as this function uses a check-effects-interactions pattern which already protects against re-entrancy: https://docs.soliditylang.org/en/v0.6.11/security-considerations.html#use-the-checks-effects-interactions-pattern